### ISSUE TITLE
Modify rule S6303: False positive when property array is built with Object.assign (AWS CDK)

### ISSUE THREAD

**Issue by gaetan-ferry-sonarsource** [SonarSource/SonarJS#3632]
# What

Rule S6303 detects RDS database objects built with the AWS CDK for which encryption is disabled. Current rule implementation analyses the `instanceProps` object passed to multiple AWS constructs to identify missing or weak encryption parameters.

The rule successfully detects true negatives when the property object is built in a very simple way. For example, the following piece of code does not raise any issue:

```typescript
const dc2 = new rds.DatabaseCluster(
    this,
    'ExplicitlyCompliantDatabaseCluster',
    {
      engine: rds.DatabaseClusterEngine.auroraMysql({ version: rds.AuroraMysqlEngineVersion.VER_2_08_1 }),
      instanceProps: {
        vpcSubnets: {
          subnetType: ec2.SubnetType.PRIVATE_WITH_EGRESS,
        },
        vpc,
      }, 
      storageEncrypted: true, // This is compliant
});
```

However, when the property object is built by aggregating multiple property arrays, false positive issues are raised. For example, the following piece of code raises an S6303 issue while the `storageEncrypted` parameter is properly set in the property object:

```typescript
var defaultProps = {
  engine: rds.DatabaseClusterEngine.auroraMysql({ version: rds.AuroraMysqlEngineVersion.VER_2_08_1 }),
  instanceProps: {
    vpcSubnets: {
      subnetType: ec2.SubnetType.PRIVATE_WITH_EGRESS,
    },
    vpc,
  },
};

const dc2 = new rds.DatabaseCluster( // False positive issue raised here
    this,
    'ExplicitlyCompliantDatabaseCluster',
    Object.assign({}, defaultProps, { 
        storageEncrypted: true,
}));
```

# Expected behavior

No issue is raised when the property object contains a valid `storageEncrypted` parameter, even if the property object is dynamically built.

# Notes

The same behavior is observed with the other constructs and encryption parameters. In all the cases, building the property object with `Object.assign` generates false positive issues.

**Comment by Wohops** [comment]
Migrated to Jira. New ticket number: JS-104
